### PR TITLE
Bug 1152486 - [Calendar] split view selector logic into its own file r=gaye

### PR DESCRIPTION
**Pull request by millermedeiros**
trying to break my app.js refactor into small commits/PRs that are easier to review and revert..
expect a bunch of these soon.

I did not add tests for the `ViewSelector` because marionette already tests it and the old logic didn't have tests anyway (and most of the logic will be replaced soon, so no point in testing it now).

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bef3533975f7b326b234c704d71bc3ff77b270fc

File: apps/calendar/js/views/view_selector.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  render: function() {
+    this._syncTodayDate();
+    this.delegate(this.element, 'click', 'a', this);
+  },
+
+  handleEvent: function(event, target) {
+    if (target.matches('[role="tab"]')) {
+      this._toggleActiveTabOnClick(event);
+    } else {
+      this._moveToDateOnClick(event);
```
Comment (gaye) [inline apps/calendar/js/views/view_selector.js]: `%s/Date/Today/` for clarity?

File: apps/calendar/js/views/view_selector.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    this.app.timeController.selectedDay = date;
+
+    event.preventDefault();
+  },
+
+  // FIXME: this should be handled by the router (Bug 1021445)
+  // Handle aria-selected attribute for tabs.
+  _toggleActiveTabOnClick: function(event) {
+    var tabs = this.element.querySelectorAll('[role="tab"]');
+    Array.from(tabs).forEach(tab => {
```
Comment (gaye) [inline apps/calendar/js/views/view_selector.js]: `Array.from` omg

File: apps/calendar/js/views/view_selector.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    this._setPresentDate();
+
+    var now = new Date();
+    var midnight = new Date(
+      now.getFullYear(), now.getMonth(), now.getDate() + 1,
+      0, 0, 0
+    );
+
+    var timeout = midnight.getTime() - now.getTime();
+    setTimeout(() => {
```
Comment (gaye) [inline apps/calendar/js/views/view_selector.js]: We should use the alarms api (or perhaps the requestSync api) for this

**Comment by gaye** [comment]
Do we have an integration test for simply toggling through the different tabs?

**Comment by millermedeiros** [comment]
@gaye all our marionette tests ends up clicking on the "month", "week" and "day" tabs. so it's already being tested

File: apps/calendar/js/views/view_selector.js (hunk 3 above)
Comment (millermedeiros) [inline apps/calendar/js/views/view_selector.js]: I'll replace this logic in the a future PR. (Bug 1118850)

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7071dc440c0379e262dce85b277256b378655dc4

**Pull request merged**